homeassistant/components/statistics/sensor.py

```diff
@@ -393,11 +389,24 @@ def _derive_unit_of_measurement(self, new_state: State) -> str | None:
     @property
     def device_class(self) -> SensorDeviceClass | None:
         """Return the class of this device."""
-        if self._state_characteristic in STAT_NUMERIC_RETAIN_UNIT:
-            _state = self.hass.states.get(self._source_entity_id)
-            return None if _state is None else _state.attributes.get(ATTR_DEVICE_CLASS)
         if self._state_characteristic in STATS_DATETIME:
             return SensorDeviceClass.TIMESTAMP
+        if self._state_characteristic in STATS_NUMERIC_RETAIN_UNIT:
+            source_state = self.hass.states.get(self._source_entity_id)
+            if source_state is None:
+                return None
+            source_device_class = source_state.attributes.get(ATTR_DEVICE_CLASS)
+            if source_device_class is None:
+                return None
+            sensor_device_class = try_parse_enum(SensorDeviceClass, source_device_class)
+            if sensor_device_class is None:
```

Comment:
Wallrus could have been used.

Reply:
Could certainly, but then black will line-break this logic into oblivion!? I already had arguably better code locally.
You wanna suggest something? Maybe you have a better idea than me

Reply:
My suggestion was already above, just wallrus it. We don't worry about formatting.

Reply:
Doesn't that contradict the wish to have easily readable and maintainable code?

I'll have a look

Reply:
Sure, that is what black is for. You dont' have to agree with its style, but it is the style we are using consistently. The formatting is maintained by black and thus not a concern.